src/main/java/ca/concordia/soen6441/project/gameplay/orders/Diplomacy.java

```diff
@@ -1,12 +1,113 @@
+/**
+ * The {@code Diplomacy} class implements the diplomacy order in the Warzone-style game.
+ * This order represents a non-aggression pact between two players for the current turn.
+ * When the order is executed, both players are marked as having a diplomacy agreement,
+ * and they will not be able to attack each other until the next turn.
+ * <p>
+ * The diplomacy card is consumed upon execution.
+ * The validate method ensures that the diplomacy action is legal.
+ */
 package ca.concordia.soen6441.project.gameplay.orders;
 
 import ca.concordia.soen6441.project.interfaces.Order;
+import ca.concordia.soen6441.project.log.LogEntryBuffer;
+import ca.concordia.soen6441.project.interfaces.Player;
+import ca.concordia.soen6441.project.gameplay.cards.DiplomacyCard;
 
 public class Diplomacy implements Order {
 
+    /**
+     * The player who issued the diplomacy order.
+     */
+    private final Player d_initiator;
+    /**
+     * The player who is the target of the diplomacy agreement.
+     */
+    private final Player d_target;
 
+    /**
+     * Constructor for Diplomacy Order.
+     *
+     * @param p_initiator The player issuing the diplomacy command.
+     * @param p_target    The target player with whom the negotiation is made.
+     */
+    public Diplomacy(Player p_initiator, Player p_target) {
+        this.d_initiator = p_initiator;
+        this.d_target = p_target;
+    }
+
+    /**
+     * Executes the Diplomacy order.
+     * <p>
+     * Steps:
+     * Validates the order using {@code validate()} method.
+     * If valid, adds each player to the other's list of negotiated players.
+     * Consumes the diplomacy card using {@code useDiplomacyCard()}.
+     * Logs the event to the log buffer.
+     */
     @Override
     public void execute() {
-        // TODO #70
+        String l_validationResult = validate();
+        if (l_validationResult != null) {
+            LogEntryBuffer.getInstance().appendToBuffer(l_validationResult, true);
+            return;
+        }
+
+        d_initiator.addNegotiatedPlayer(d_target);
+        d_target.addNegotiatedPlayer(d_initiator);
+
+        // Remove the used DiplomacyCard
+        d_initiator.getHandOfCardsManager().getDiplomacyCardManager().removeCard();

```

Comment:
Card shouldn't be removed on the execute, it should be removed once the order is issued.